endpoint.go

```diff
@@ -302,11 +305,33 @@ func handleIncoming(e *endPoint) func(http.ResponseWriter, *http.Request) {
 				out = e.caller.Do(ref)
 				// fmt.Print("!")
 			}
+			if pub == "1" { //wrap output of public api
+				WrapResult(e.caller.outParams, &out)
+			}
 			writeOutput(w, e.caller.outParams, out, e.info.Pretty)
 		}
 	}
 }
 
+func WrapResult(outType []string, outVals *[]reflect.Value) {
+
+	wrapper := make(map[string]interface{})
+
+	if len(outType) == 1 && outType[0] == "map" {
```

Comment:
if and else block contains same code, just index value is not same. remove duplicate code.